TransformationPlayground/playground_frontend/src/app/page.tsx

```diff
@@ -0,0 +1,206 @@
+
+        setTransformLogic(data.transform_logic || "");
+        setOutputShape(prettyOutputShape || "");
+        setValidationReport(formattedValidationReport || "");
+      } else {
+        console.error("Failed to fetch recommendation");
+      }
+    } catch (error) {
+      console.error("Error:", error);
+
+      if (error instanceof SyntaxError) {
+        alert("Invalid JSON: Please check your input and try again.");
+      }
+    } finally {
+      setIsLoading(false); // Stop visual spinner
+    }
+  };
+
+  return (
+    <Grid
+      gridDefinition={[
+        { colspan: 4 }, // Left Column
+        { colspan: 4 }, // Middle Column
+        { colspan: 4 }, // Right Column
+      ]}
+    >
+      {/* User Input Column */}
+      <Container header="User Input Panel">
+        <FormField
+          label="Original JSON Input"
+          description="Paste or edit the JSON blob that represents the Elasticsearch 6.8 Index JSON."
+        >
+          <Textarea
+            value={inputShape}
+            onChange={({ detail }) => setInputShape(detail.value)}
+            placeholder='{"index_name": "my-index", "index_json": { index settings and mappings }}'
+            rows={30}
+          />
+        </FormField>
+      </Container>
+
+      {/* Transformation Column */}
+      <Container header="Transformation Panel">
+        <FormField label="Transformation Settings">
+          <Grid
+            gridDefinition={[
+              { colspan: 6 }, // Source Version Dropdown
+              { colspan: 6 }, // Target Version Dropdown
+              { colspan: 6 }, // Transform Type Dropdown
+              { colspan: 6 }, // Transform Language Dropdown
+            ]}
+          >
+            <Select
+              selectedOption={sourceVersion}
+              onChange={({ detail }) => setSourceVersion(detail.selectedOption!)} 
+              options={sourceVersionOptions}
+              placeholder="Select Source Version"
+            />
+            <Select
+              selectedOption={targetVersion}
+              onChange={({ detail }) => setTargetVersion(detail.selectedOption!)}
+              options={targetVersionOptions}
+              placeholder="Select Target Version"
+            />
+            <Select
+              selectedOption={transformType}
+              onChange={({ detail }) => setTransformType(detail.selectedOption!)}
+              options={transformTypeOptions}
+              placeholder="Select Transform Type"
+            />
+            <Select
+              selectedOption={transformLanguage}
+              onChange={({ detail }) => setTransformLanguage(detail.selectedOption!)}
+              options={transformLanguageOptions}
+              placeholder="Select Language"
+            />
+          </Grid>
+        </FormField>
+        <br />
+        <Button onClick={handleGetRecommendation}>
+          {isLoading ? <Spinner/> : "Get GenAI Recommendation"}
```

Comment:
Can you use this tool without getting an API recommendation but manually creating the transformation logic?

Reply:
Yep, that's one of the next things I want to implemented (should be real quick).  Basically, make the text area for the transformation writable and add buttons for "Test" and "Deploy" to the righthand panel.  "Test" would take the transform logic and run it through the same validation process we apply to the LLM output.  "Deploy" would basically do the same thing as "Test" but without the final cleanup step where we do stuff like delete the index off the test target.  This enables the user to use the Playground for small-scale migration efforts directly without needing the full Migration Assistant, as well as probably setting up the Playground to be how we do metadata migrations in general.